### PR TITLE
chore: update requests to 2.32.2

### PR DESCRIPTION
**Pull request by JoeWang1127**
Resolve https://github.com/googleapis/google-cloud-java/security/dependabot/78

### CODE REVIEW

File: .github/requirements.txt [hunk 1]
```diff
@@ -1,26 +1 @@
...
-    # via requests
-requests==2.31.0 \
-    --hash=sha256:58cd2187c01e70e6e26505bca751777aa9f2ee0b7f4300988b709f44e013003f \
-    --hash=sha256:942c5a758f98d790eaed1a29cb6eefc7ffb0d1cf7af05c3d2791656dbd6ad1e1
-    # via -r requirements.in
-urllib3==1.26.18 \
-    --hash=sha256:34b97092d7e0a3a8cf7cd10e386f401b3737364026c45e622aa02903dffe0f07 \
-    --hash=sha256:f8ecc1bba5667413457c529ab955bf8c67b45db799d159066261719e328580a0
-    # via requests
+requests==2.32.2
```
Comment (blakeli0) [inline .github/requirements.txt:1]: Is there any scripts still use it? I think it was used by `new-client.py`, but now that we moved to hermetic build, we may not need it anymore?

File: .github/requirements.in
Comment (blakeli0) [inline .github/requirements.in:1]: If we do still need `requests`, we should keep this `requiremenst.in` file. It was introduced to mitigate a security issue. @mpeddada1 probably has more info regarding why we need it.

Comment (JoeWang1127) [reply .github/requirements.in:1]: Sure, I'll let @mpeddada1 to review this pr.

Comment (JoeWang1127) [reply .github/requirements.in:1]: Do we need two files (`requirement.in` and `requirement.txt`) to lock the dependency version?

File: .github/requirements.txt (hunk 1 above)
Comment (JoeWang1127) [reply .github/requirements.txt]: Answered in https://github.com/googleapis/google-cloud-java/pull/10992#discussion_r1657305228

**Review by blakeli0 — APPROVED** [review]

File: .github/requirements.in
Comment (mpeddada1) [reply .github/requirements.in:1]: Sorry for missing this earlier! Technically, even though requirements.txt is directly used in the github actions job, `requirements.in` is used to manage the high level version specifications. It is used to re-generate the `requirements.txt` through `pip-compile --generate-hashes .github/requirements.in`. This allows us to update the dependency in one place and let the re-generation take care of updating the dependency and all of its deps in `requirements.txt`.

**Pull request merged**